spec/features/delete_taxon_spec.rb

```diff
@@ -0,0 +1,142 @@
+require "rails_helper"
+
+RSpec.feature "Delete Taxon", type: :feature do
+  include ContentItemHelper
+  include PublishingApiHelper
+
+  scenario "a taxon with no children" do
+    given_a_taxon_with_no_children
+    when_i_visit_the_taxon_page
+    and_i_click_to_delete_the_taxon
+    then_i_expect_to_be_informed_that_im_about_to_delete_the_taxon
+    and_i_should_see_the_action_buttons
+    and_i_click_delete
+    then_i_expect_the_request_to_have_been_made
+  end
+
+  scenario "a parent taxon with children" do
+    given_a_taxon_with_children
+    when_i_visit_the_taxon_page
+    then_i_expect_to_see_the_child_taxon
+    and_i_click_to_delete_the_taxon
+    then_i_should_see_a_warning_message
+    and_i_should_see_the_action_buttons
+    and_i_click_delete
+    then_i_expect_the_request_to_have_been_made
+  end
+
+  def given_a_taxon_with_no_children
+    @content_id = SecureRandom.uuid
+
+    @taxon = {
+      content_id: @content_id,
+      title: 'Taxon 1',
+      description: 'A description',
+      base_path: 'A base path',
+      publication_state: 'State',
+      document_type: 'taxon',
+      details: {
+        internal_name: 'Internal name',
+        notes_for_editors: 'Notes for editors',
+      }
+    }
+
+    publishing_api_has_taxons(
+      [@taxon],
+      page: 1,
+      per_page: 2,
+    )
+
+    publishing_api_has_item(@taxon)
+
+    publishing_api_has_links(
+      content_id: @content_id,
+      links: {
+        topics: [],
+        parent_taxons: []
+      }
+    )
+
+    publishing_api_has_linked_content_items(@content_id, "taxons", [@taxon])
+
+    publishing_api_has_expanded_links(
+      content_id: @content_id,
+      expanded_links: {
+        documents: []
+      }
+    )
+  end
+
+  def a_child_taxon
+    @child_taxon_content_id = SecureRandom.uuid
+    @child_taxon = {
```

Comment:
We could use the helper method I mentioned above to generate this: https://github.com/alphagov/content-tagger/blob/master/spec/support/content_item_helper.rb#L2